src/streamsync/ui_manager.py

```diff
@@ -137,6 +136,13 @@ def _create_component(component_tree: ComponentTree,  component_type: str, **kwa
     raw_content: dict = kwargs.pop("content", {})
     content = {key: _prepare_value(value) for key, value in raw_content.items()}
 
+    if component_type == "page" and "id" not in kwargs:
```

Comment:
Add a comment block that describe why we set automatic id on Page and not on other component